src/web/metrics.rs

```diff
@@ -130,6 +130,7 @@ pub static MAX_DB_CONNECTIONS: Lazy<IntGauge> = Lazy::new(|| {
 });
 
 #[cfg(not(windows))]
+#[allow(dead_code)]
```

Comment:
clippy failed without these on local runs, not sure why it's not failing in GitHub actions running it.

Reply:
Instead of this I'd change the [`#[cfg(target_os = "linux")]`](https://github.com/rust-lang/docs.rs/blob/525c6f855571962bd594b659f0da552dfbb3146a/src/web/metrics.rs#L171) to `#[cfg(not(windows))]`

Reply:
@Kixiron sure I'll change that.

What about the `dead_code` it's OK to add it?

Reply:
Changing to `#[cfg(target_os = "linux")]` make my IDE (InteliJ) grey it out since I'm on a Mac.

Which also has file descriptors, so maybe keeping the `not windows` is OK?

Reply:
No, that should be able to be removed, we don't want it lying around if we don't actually need it

Reply:
That can't be `cfg(not(windows))` since `procfs` is a linux only thing (it'd fail on macOS). This should probably be `#[cfg(target_os = "linux")]` as well.

Reply:
I actually think it's a false alarm , I found a couple of threads about similar issues.

> src/web/metrics.rs:134:pub static OPEN_FILE_DESCRIPTORS: Lazy<IntGauge> = Lazy::new(|| {
> src/web/metrics.rs:176:        OPEN_FILE_DESCRIPTORS.set(process.fd().unwrap().len() as i64);

> src/web/metrics.rs:144:pub static CURRENTLY_RUNNING_THREADS: Lazy<IntGauge> = Lazy::new(|| {
> src/web/metrics.rs:177:        CURRENTLY_RUNNING_THREADS.set(process.stat().unwrap().num_threads as i64);

Reply:
On non-linux and non-windows systems the static exists but the invocation doesn't, please change the `cfg`s on the statics to `#[cfg(target_os = "linux")]` and remove the `#[allow(dead_code)]`s, that should fix things

Reply:
Yeah just noticed and pushed it, the declaration and usage had different conditions.